gramps/gen/db/generic.py

```diff
@@ -2739,3 +2739,73 @@ def set_serializer(self, serializer_name):
             self.serializer = BlobSerializer
         elif serializer_name == "json":
             self.serializer = JSONSerializer
+
+    def select_from_table(
+        self, table_name, what=None, where=None, order_by=None, env=None
```

Comment:
I know that currently you only pass a single table_name but if you were to make this low level method take an array of table_names then I think everything exists to do cross table selects.
With that it would be possible to add a generic filter UI that allows the user to enter a query defined in terms of tables, what, where and order_by strings. I readily acknowledge that this might be a more "advanced user" feature but it would be *really* powerful.
It will also allow code to execute more complex queries directly within the DB ❤️

Reply:
@stevenyoungs, I like this idea! First, let's make sure that we can get this type of method added, and then think about expanding it.

(I'd like to see an idea of how you would represent a JOIN, but I don't want you to spend too much time if this PR gets rejected.)